components/feature/toolbar/src/main/java/mozilla/components/feature/toolbar/ToolbarAutocompleteFeature.kt

```diff
@@ -0,0 +1,28 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.feature.toolbar
+
+import mozilla.components.concept.toolbar.AutocompleteProvider
+import mozilla.components.concept.toolbar.Toolbar
+
+/**
+ * Feature implementation for connecting a toolbar with a list of autocomplete providers.
+ */
+class ToolbarAutocompleteFeature(val toolbar: Toolbar, private val providers: List<AutocompleteProvider>) {
+    init {
+        toolbar.setAutocompleteListener { value, acceptor ->
+            acceptor?.let {
```

Comment:
I wonder if we should ever end up here with a `null` acceptor. This seems to be unneeded work. I remember that this may come from InlineAutocomplete; I wonder if we should stop that event ("autocomplete for nobody") as early as possible. Doesn't need to be here though.

Reply:
Yeah, that crept in from some existing code, and I didn't clean it up.